third-party/rspamd/src/main/java/org/apache/james/rspamd/RSpamDScanner.java

```diff
@@ -35,26 +36,42 @@
 import org.apache.mailet.Mail;
 import org.apache.mailet.PerRecipientHeaders;
 import org.apache.mailet.base.GenericMailet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import com.google.common.collect.ImmutableList;
 
 public class RSpamDScanner extends GenericMailet {
     public static final AttributeName FLAG_MAIL = AttributeName.of("org.apache.james.rspamd.flag");
     public static final AttributeName STATUS_MAIL = AttributeName.of("org.apache.james.rspamd.status");
+    private static final Logger LOGGER = LoggerFactory.getLogger(RSpamDScanner.class);
 
     private final RSpamDHttpClient rSpamDHttpClient;
+    private Optional<String> spamProcessor;
 
     @Inject
     public RSpamDScanner(RSpamDHttpClient rSpamDHttpClient) {
         this.rSpamDHttpClient = rSpamDHttpClient;
     }
 
+    @Override
+    public void init() {
+        spamProcessor = getInitParameterAsOptional("spamProcessor");
+    }
+
     @Override
     public void service(Mail mail) throws MessagingException {
         AnalysisResult rSpamDResult = rSpamDHttpClient.checkV2(new MimeMessageInputStream(mail.getMessage())).block();
 
         mail.getRecipients()
             .forEach(recipient -> appendRSpamDResultHeader(mail, recipient, rSpamDResult));
+
+        if (rSpamDResult.getHasVirus()) {
+            if (isDebug()) {
+                LOGGER.debug("Detected a mail containing virus. Sending mail {} to {}", mail, spamProcessor);
```

Comment:
You should not be logging this if spamProcessor is not specified: this is very confusing...

Reply:
Also I'm in favor of always adding this helpful debug log, when a virus is detected: this is very valuable information! I would go as far as logging at the INFO level.